liquidctl/driver/kraken3.py

```diff
@@ -648,7 +648,7 @@ def _find_winusb_device(self, vid, pid, serial):
         return False
 
     def _get_fw_version(self, clear_reports=True):
-        if self.fw is not None:
+        if hasattr(self, "fw") and self.fw is not None:
```

Comment:
How about initializing the driver with `self._fw = None` instead?

---

On a somewhat related note, do we want this field to be public or private (in terms of Python's conventions)?

In the former case, it should have a more precise and abbreviation-free name like `self.firmware_version`, and we should be mindful that once it's public, it'd be subject to our API stability promises.

In the latter case, it just need to be prepended with a underscore: `self._fw`.

Reply:
> How about initializing the driver with self._fw = None instead?

Pushed that now. Debated between that and hasattr, init to None is probably more 'pythonic'.

As for it being public, that seems OK to me...? It's always read in the init method. If you want I'll add that.

Reply:
I think it's better to keep it private: we don't want to promise that this will always be available, and in this format, and it would also only be available in this driver.